csharp-intermediate-topics/WhenToUseReaderWriterLockSlimOverLockInCSharp/WhenToUseReaderWriterLockSlimOverLockInCSharp/BaseReaderWriter.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Diagnostics;
+
+namespace WhenToUseReaderWriterLockSlimOverLockInCSharp
+{
+    public abstract class BaseReaderWriter
+    {
+        private readonly Stopwatch _stopwatch = new();
+        protected readonly List<int> _listOfNumbers = new();
+
+        public List<int> ListOfNumbers
+        {
+            get { return _listOfNumbers; }
+        }
+
+        protected int GetNextRandom()
+        {
+            var maxNumber = _listOfNumbers.Count > 0 ? _listOfNumbers.Count - 1 : 0;
+            return Random.Shared.Next(0, maxNumber);
+        }
+
+        public long Execute(ThreadExecutionConfiguration config)
+        {
+            var threads = new List<Thread>();
+
+            for (var cnt = 0; cnt < config.ReaderThreadsCount; cnt++)
+            {
+                var readThread = new Thread(ReadListCount);
+                threads.Add(readThread);
+            }
+
```

Comment:
I did see your comment about sticking with `Thread` because you are using the word thread in the article. I did a quick scan of the article, and it appears that outside of the code block, you are using the word thread, referring to a principle of CS, rather than the concrete `System.Threading.Thread` class. With that being the case, we both modernize and simplify the code by using `Task`. Again, you can look at the fiddle from my previous review and see how we can use `Task` and how this simplifies our code. It also allows us to provide concrete arguments to our methods that will be executed, thus providing additional type safety.

Reply:
changed to Task